src/utils/update_lattices.py

```diff
@@ -0,0 +1,201 @@
+    # If that was unsuccessful, attempt to randomly guess a new id number.
+    for i in range(max_random_attempts):
+        num = randint(1, 2147483647)
+        if num not in current_inds:
+            return num
+
+    # Raise an error if an id was not found.
+    raise RuntimeError('Could not find a unique id number for a new universe.')
+
+
+def get_lat_id(lattice_element):
+    """Return the id integer of the lattice_element."""
+    assert isinstance(lattice_element, ET.Element)
+    if 'id' in lat.attrib:
+        return int(lat.attrib['id'].strip())
+    elif any([child.tag == 'id' for child in lat]):
+        elem = lat.find('id')
+        return int(elem.text.strip())
+    else:
+        raise RuntimeError('Could not find the id for a lattice.')
+
+
+def pop_lat_outside(lattice_element):
+    """Return lattice's outside material and remove from attributes/elements."""
+    assert isinstance(lattice_element, ET.Element)
+
+    # Check attributes.
+    if 'outside' in lat.attrib:
+        material = lat.attrib['outside'].strip()
+        del lat.attrib['outside']
+
+    # Check subelements.
+    elif any([child.tag == 'outside' for child in lat]):
+        elem = lat.find('outside')
+        material = elem.text.strip()
+        lat.remove(elem)
+
+    # No 'outside' specified.  This means the outside is a void.
+    else:
+        material = 'void'
+
+    return material
+
+
+if __name__ == '__main__':
+    args = parse_args()
+    for fname in args.input:
+        # Parse the XML data.
+        tree = ET.parse(fname)
+        root = tree.getroot()
+
+        # Ignore files that do not contain lattices.
+        if all([child.tag != 'lattice' for child in root]): continue
+
+        # Get a set of already-used universe and cell ids.
+        uids = get_universe_ids(root)
+        cids = get_cell_ids(root)
+        taken_ids = uids.union(cids)
+
+        # Update the definitions of each lattice
+        for lat in root.iter('lattice'):
+            # Get the lattice's id.
+            lat_id = get_lat_id(lat)
+
+            # Pop the 'outside' material.
+            material = pop_lat_outside(lat)
+
+            # Get an id number for a new outer universe.  Ideally, the id should
+            # be close to the lattice's id.
+            new_uid = find_new_id(taken_ids, preferred=lat_id)
+            assert new_uid not in taken_ids
+
+            # Add the new universe filled with the old 'outside' material to the
+            # geometry.
+            new_cell = ET.Element('cell')
+            new_cell.attrib['id'] = str(new_uid)
+            new_cell.attrib['universe'] = str(new_uid)
+            new_cell.attrib['material'] = material
+            root.append(new_cell)
+            taken_ids.add(new_uid)
+
+            # Add the new universe to the lattice's 'outer' attribute.
+            lat.attrib['outer'] = str(new_uid)
+
+        # Move the original geometry file to preserve it.
+        move(fname, ''.join((fname, '.original')))
```

Comment:
The `+` concatenation will work fine here.  I am in the habit of using `''.join()` because it is often faster.  But there won't be any speed improvement here so I'll switch back to `+` for readability in the next commit.